numpy/core/include/numpy/npy_common.h

```diff
@@ -10,6 +10,14 @@
 #include <npy_config.h>
 #endif
 
+// int*, int64* should be propertly aligned on ARMv7 to avoid bus error
+#if !defined(NPY_STRONG_ALIGNMENT) && defined(__arm__) && !(defined(__aarch64__) || defined(_M_ARM64))
+#define NPY_STRONG_ALIGNMENT 1
+#endif
+#if !defined(NPY_STRONG_ALIGNMENT)
+#define NPY_STRONG_ALIGNMENT 0
+#endif
```

Comment:
I just noticed this addition. Is this the same (idea) as `NPY_CPU_HAVE_UNALIGNED_ACCESS`, which we do already use in a handful of places?  I am a bit worried of duplicating this logic, especially since this seems to assume that all CPUs have unaligned access instead of the opposite of assuming x86 and amd64 always supporting unaligned access?

Reply:
Thanks for pointing this out, in this case, only `armv7 `needs aligned access, which assume that other CPUs supporting unaligned access, so `NPY_CPU_HAVE_UNALIGNED_ACCESS ` and `NPY_STRONG_ALIGNMENT ` are like two sides of a coin, you are right about the duplicating logic, will integrate together in follow-up PR.